### PR TITLE
Add selenium firefox container

### PR DESCRIPTION
**Pull request by sydoluciani**
added docker-py dependency and passed environment variables in tox.ini that is going to be used in Deformdemo to start selenium container.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -72,11 +72,13 @@ commands =
 # interpreter only
 [testenv:functional3]
 # Allow override test browser
-passenv = WEBDRIVER DISPLAY FIREFOX_PATH
+passenv = DISPLAY WEBDRIVER DISPLAY FIREFOX_PATH TRAVIS URL WAITTOSTART CONTAINERTZ
```
Comment (stevepiercy) [inline tox.ini:75]: Why is DISPLAY here twice?

**Comment by stevepiercy** [comment]
Is this a draft PR?

**Comment by sydoluciani** [comment]
> Is this a draft PR?

It is a working and functional PR, you can clone, set the environment variable and verify the docker container start, stop and test completes successfully. I might make new changes tomorrow, do you want me to create a new branch or just push it to this branch ?

**Comment by stevepiercy** [comment]
I'll try it out, and you can set the PR to "Draft" so that it does not get merged.

**Comment by sydoluciani** [comment]
@stevepiercy I made all requested changes that you made in other pull request, please check and if you can please test it locally and let me know.

**Comment by sydoluciani** [comment]
@stevepiercy I did update the contributing.md but needs some indentation, I will fix it tomorrow along other required changes.

**Review by stevepiercy — COMMENTED** [review]
I'm going to push some changes to your PR, and hope that it updates. Mostly grammar stuff.

**Review by stevepiercy — APPROVED** [review]
This is really cool. Thank you!

I still have a few more things to reorganize in contributing.md, but this all works. I can toggle between how I want to run `functional3` tests by setting `WEBDRIVER`.

I'd appreciate it if you would take one last look at the diff on contributing.md, especially the one piece about setting the `URL` environment variable.

**Comment by sydoluciani** [comment]
> This is really cool. Thank you!
> 
> I still have a few more things to reorganize in contributing.md, but this all works. I can toggle between how I want to run `functional3` tests by setting `WEBDRIVER`.
> 
Thank you for fixing the indentation, my development server is running on AWS and can't properly check the formating before pushing to github. 
> I'd appreciate it if you would take one last look at the diff on contributing.md, especially the one piece about setting the `URL` environment variable.

URL should be host name or ip address. [URL value](https://github.com/Pylons/deformdemo/blob/master/deformdemo/test.py#L35) is being sent to browser to connect to server that has been started using pserve, putting 0.0.0.0 in browser wont work. 

Not sure why localhost didn't work for me, even though my development server is not local to me and it is running on AWS, but it is local to selenium docker container. as longs as contributors won't puzzle with first connection failure and try the ip address or hostname instead of local host, we are good.

**Comment by stevepiercy** [comment]
Thank you! Change made, I'll wait for build to pass before merging.

**Pull request merged**